pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -1213,24 +1267,33 @@ func TestReconcileInvalidTaskRuns(t *testing.T) {
 			defer cancel()
 			c := testAssets.Controller
 			clients := testAssets.Clients
-			err := c.Reconciler.Reconcile(context.Background(), getRunName(tc.taskRun))
+			reconciler := c.Reconciler.(*Reconciler)
+			fr := reconciler.Recorder.(*record.FakeRecorder)
+			err := reconciler.Reconcile(context.Background(), getRunName(tc.taskRun))
+
 			// Events are sent in a goroutine, let's sleep a bit to make sure they're
 			// captured by the fake client-go action list
-			time.Sleep(100 * time.Millisecond)
+			time.Sleep(2000 * time.Millisecond)
```

Comment:
2 * time.Second?

I'd really like to take some time to think about whether or not we can avoid having sleeps in these tests, they just always seem to lead to pain later on

Reply:
So, actually this was needed with the fake client-go and checking of actions, and it's one of the reason I looked for another way of running the test.

Now, since I switched to the fake recorder and the `checkEvents` approach, my bet is that this is not required anymore. I will try removing it.

Reply:
nice!!